stan/math/rev/fun/quad_form.hpp

```diff
@@ -20,13 +20,13 @@ class quad_form_vari_alloc : public chainable_alloc {
   inline void compute(const Eigen::Matrix<double, Ra, Ca>& A,
                       const Eigen::Matrix<double, Rb, Cb>& B) {
     matrix_d Cd = B.transpose() * A * B;
+    if (sym_) {
+      matrix_d M = 0.5 * (Cd + Cd.transpose());
+      Cd = M;
```

Comment:
```
Cd = 0.5 * (Cd + Cd.transpose()).eval();
```

Reply:
Ah... I missed the `.eval()`; I did that first (without the `.eval()`) and got non-sense, so I just assigned it to a temporary and moved on. 

I can open a new PR to change that.

Reply:
It's fine now, this does the same thing.